test/e2e/pivoting.go

```diff
@@ -0,0 +1,447 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	bmov1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+
+	dockerTypes "github.com/docker/docker/api/types"
+	docker "github.com/docker/docker/client"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/config"
+	framework "sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	bmoPath                    = "BMOPATH"
+	ironicTLSSetup             = "IRONIC_TLS_SETUP"
+	ironicBasicAuth            = "IRONIC_BASIC_AUTH"
+	KIND                       = "kind"
+	NAMEPREFIX                 = "NAMEPREFIX"
+	restartContainerCertUpdate = "RESTART_CONTAINER_CERTIFICATE_UPDATED"
+	ironicNamespace            = "IRONIC_NAMESPACE"
+	isTrue                     = "true" // golint string `true` has 3 occurrences, make it a constant (goconst)
+)
+
+type PivotingInput struct {
+	E2EConfig             *clusterctl.E2EConfig
+	BootstrapClusterProxy framework.ClusterProxy
+	TargetCluster         framework.ClusterProxy
+	SpecName              string
+	ClusterName           string
+	Namespace             string
+	ArtifactFolder        string
+	ClusterctlConfigPath  string
+}
+
+func pivoting(ctx context.Context, inputGetter func() PivotingInput) {
+	Logf("Starting pivoting tests")
+	input := inputGetter()
+	numberOfWorkers := int(*input.E2EConfig.GetInt32PtrVariable("WORKER_MACHINE_COUNT"))
+	numberOfControlplane := int(*input.E2EConfig.GetInt32PtrVariable("CONTROL_PLANE_MACHINE_COUNT"))
+	numberOfAllBmh := numberOfWorkers + numberOfControlplane
+
+	listBareMetalHosts(ctx, input.BootstrapClusterProxy.GetClient(), client.InNamespace(input.Namespace))
+	listMetal3Machines(ctx, input.BootstrapClusterProxy.GetClient(), client.InNamespace(input.Namespace))
+	listMachines(ctx, input.BootstrapClusterProxy.GetClient(), client.InNamespace(input.Namespace))
+	listNodes(ctx, input.TargetCluster.GetClient())
+
+	By("Remove Ironic containers from the source cluster")
+	ephemeralCluster := os.Getenv("EPHEMERAL_CLUSTER")
+	isDeployment := isTrue
+	if ephemeralCluster == KIND {
+		isDeployment = "false"
+	}
+	removeIronic(ctx, func() RemoveIronicInput {
+		return RemoveIronicInput{
+			ManagementCluster: input.BootstrapClusterProxy,
+			IsDeployment:      isDeployment,
+			Namespace:         input.E2EConfig.GetVariable(ironicNamespace),
+			NamePrefix:        input.E2EConfig.GetVariable(NAMEPREFIX),
+		}
+	})
+
+	By("Create Ironic input.Namespace")
```

Comment:
A bit too much find/replace maybe :joy: 
```suggestion
	By("Create Ironic namespace")
```

Reply:
thanks for catching that